File: Easy/C++/Check If It Is a Straight Line(1232).cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {

	bool res = false;
    float slope, curSlope;
    sort(coordinates.begin(), coordinates.end());
    for(int i=1; i<coordinates.size();i++)
    {
        curSlope = (float)(coordinates[i][1]-coordinates[i-1][1])/(float)(coordinates[i][0]-coordinates[i-1][0]);
        if (i == 1)
            slope = curSlope;
        else if (curSlope != slope)
            break;
        
        if (i == coordinates.size()-1)
            res = true;
    }
    
    return res;
}
};

File: Easy/Python 3/Check If It Is a Straight Line(1232).py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        def get_slope(a, b):
            if b[0] == a[0]:
                return '#'
            return (b[1]-a[1])/(b[0]-a[0])
        slope = get_slope(coordinates[0], coordinates[1])
        for i in range(1, len(coordinates)-1):
            if get_slope(coordinates[i], coordinates[i+1]) != slope:
                return False
        return True